File: Trees/BinaryTrees/DistanceBetweenTwoNodes.java
package Trees.BinaryTrees;

public class DistanceBetweenTwoNodes {
    private static TreeNode getLCA(TreeNode root, int l1, int l2) {
        if (root == null) return null;

        if (root.data == l1 || root.data == l2) return root;

        TreeNode left = getLCA(root.left, l1, l2);
        TreeNode right = getLCA(root.right, l1, l2);

        if (left != null && right != null)
            if ((left.data == l1 && right.data == l2) || (left.data == l2 && right.data == l1)) return root;
        return left != null ? left : right;
    }

    private static int findNodeFromLCA(TreeNode root, int target) {
        if (root == null) return -1;

        if (root.data == target) return 0;

        int left = findNodeFromLCA(root.left, target);
        int right = findNodeFromLCA(root.right, target);

        if (left != -1) return left + 1;
        if (right != -1) return right + 1;
        return left;
    }

    public static void main(String[] args) {
        TreeNode root = TreeNode.getInstance2();
        TreeNode LCA = getLCA(root, 8, 11);
        int eightDistance = findNodeFromLCA(LCA, 8);
        int elevenDistance = findNodeFromLCA(LCA, 11);
        System.out.println("8 : " + eightDistance + "\n 11 : " + elevenDistance);
    }
}

File: Trees/BinaryTrees/DistanceBetweenTwoNodes.cpp
#include <bits/stdc++.h>
using namespace std;

typedef struct TreeNode{
    int val;
    TreeNode* left,*right;

    TreeNode(){
    }

    TreeNode(int val){
        this->val = val;
        left = right = NULL;
    }
}TreeNode;

class Util{
	public:
    static TreeNode* getTree(){
        TreeNode* root = new TreeNode(10);
        root->left = new TreeNode(20);
        root->right = new TreeNode(30);

        root->left->left = new TreeNode(40);
        root->left->right = new TreeNode(50);
        root->left->right->left = new TreeNode(60);
        root->left->right->right = new TreeNode(70);
        return root;
    }

    static TreeNode* getLCA(TreeNode* root,int t1,int t2){
        if(root == nullptr) return root;

        if(root->val == t1 || root->val == t2)  return root->val;

        int left = getLCA(root->left,t1,t2);
        int right = getLCA(root->right,t1,t2);

        if(left == nullptr || right == nullptr) return left != nullptr ? left : right;

        if((left == t1 && right == t2) || (left == t2 && right == t1))  return root->val;

        return left != nullptr ? left : right;
    }

    static int getDistanceFromRoot(TreeNode* root,int target){
        if(root == nullptr) return -1;

        if(root->val == target) return 0;

        int left = getDistanceFromRoot(root->left,target);
        int right = getDistanceFromRoot(root->right,target);

        if(left != -1)  return left + 1;
        if(right != -1) return right + 1;

        return left;
    }

};
int main(){
    TreeNode* root = Util::getTree();
    TreeNode* lca = Util::getLCA(root,70,30);
    int dist1 = Util::getDistanceFromRoot(lca,70);
    int dist2 = Util::getDistanceFromRoot(lca,30);

    cout<<(dist1 + dist2);

    return 0;
}